tests/integration-tests/conftest.py

```diff
@@ -855,7 +858,11 @@ def cfn_stacks_factory(request):
     factory = CfnStacksFactory(request.config.getoption("credential"))
     yield factory
     if not request.config.getoption("no_delete"):
-        factory.delete_all_stacks()
+        if request.config.getoption("retain_ad_stack"):
```

Comment:
Minor: we can simplify and make it ready for extensions.

```
excluded_stacks = []
if request.config.getoption("retain_ad_stack"):
   excluded_stacks.append(.....)
factory.delete_all_stacks(excdluded_stacks)
```

In this way you avoid repetition of the delate_all_Stacks function and when we will add a new retention it will only be a matter of adding elements in the excluded_stacks